src/main/java/org/apache/maven/plugins/dependency/ListClassesMojo.java

```diff
@@ -0,0 +1,391 @@
+            for ( String repo : repos )
+            {
+                repoList.add( parseRepository( repo, always ) );
+            }
+        }
+
+        ProjectBuildingRequest buildingRequest =
+                new DefaultProjectBuildingRequest( session.getProjectBuildingRequest() );
+
+        Settings settings = session.getSettings();
+        repositorySystem.injectMirror( repoList, settings.getMirrors() );
+        repositorySystem.injectProxy( repoList, settings.getProxies() );
+        repositorySystem.injectAuthentication( repoList, settings.getServers() );
+
+        buildingRequest.setRemoteRepositories( repoList );
+
+        return buildingRequest;
+    }
+
+    protected ArtifactCoordinate toArtifactCoordinate( DependableCoordinate dependableCoordinate )
+    {
+        ArtifactHandler artifactHandler = artifactHandlerManager.getArtifactHandler( dependableCoordinate.getType() );
+        DefaultArtifactCoordinate artifactCoordinate = new DefaultArtifactCoordinate();
+        artifactCoordinate.setGroupId( dependableCoordinate.getGroupId() );
+        artifactCoordinate.setArtifactId( dependableCoordinate.getArtifactId() );
+        artifactCoordinate.setVersion( dependableCoordinate.getVersion() );
+        artifactCoordinate.setClassifier( dependableCoordinate.getClassifier() );
+        artifactCoordinate.setExtension( artifactHandler.getExtension() );
+        return artifactCoordinate;
+    }
+
+    ArtifactRepository parseRepository( String repo, ArtifactRepositoryPolicy policy )
+            throws MojoFailureException
+    {
+        // if it's a simple url
+        String id = "temp";
+        ArtifactRepositoryLayout layout = getLayout( "default" );
+
+        // if it's an extended repo URL of the form id::layout::url
+        if ( repo.contains( "::" ) )
+        {
+            Matcher matcher = ALT_REPO_SYNTAX_PATTERN.matcher( repo );
+            if ( !matcher.matches() )
+            {
+                throw new MojoFailureException( repo, "Invalid syntax for repository: " + repo,
+                        "Invalid syntax for repository. Use \"id::layout::url\" or \"URL\"." );
+            }
+
+            id = matcher.group( 1 ).trim();
+            if ( !( matcher.group( 2 ) == null || matcher.group( 2 ).trim().isEmpty() ) )
+            {
+                layout = getLayout( matcher.group( 2 ).trim() );
+            }
+            repo = matcher.group( 3 ).trim();
+        }
+        return new MavenArtifactRepository( id, repo, layout, policy, policy );
+    }
+
+    private ArtifactRepositoryLayout getLayout( String id )
+            throws MojoFailureException
+    {
+        ArtifactRepositoryLayout layout = repositoryLayouts.get( id );
+
+        if ( layout == null )
+        {
+            throw new MojoFailureException( id, "Invalid repository layout", "Invalid repository layout: " + id );
+        }
+
+        return layout;
+    }
+
+    /**
+     * @return {@link #skip}
```

Comment:
I don't think you need these getter methods, and probably not the setters

Reply:
fixed